### PR TITLE
feat(sell-bag): closing link sell bag open/close state with sell mode activation

### PR DESCRIPTION
**Pull request by JFrankfurt**
https://uniswaplabs.atlassian.net/browse/WEB-1945

### CODE REVIEW

File: src/components/NavBar/ShoppingBag.tsx [hunk 1]
```diff
@@ -2,14 +2,33 @@ import { NavIcon } from 'components/NavBar/NavIcon'
...
+import { useBag, useSellAsset } from 'nft/hooks'
+import { useCallback, useEffect, useState } from 'react'
+import shallow from 'zustand/shallow'
 
 export const ShoppingBag = () => {
   const itemsInBag = useBag((state) => state.itemsInBag)
   const [bagQuantity, setBagQuantity] = useState(0)
 
-  const toggleBag = useBag((s) => s.toggleBag)
+  const { bagExpanded, setBagExpanded } = useBag(
```
Comment (vm) [inline src/components/NavBar/ShoppingBag.tsx:13]: this part i'm not actually following

File: src/components/NavBar/ShoppingBag.tsx [hunk 2]
```diff
@@ -2,14 +2,33 @@ import { NavIcon } from 'components/NavBar/NavIcon'
...
+    ({ bagExpanded, setBagExpanded }) => ({ bagExpanded, setBagExpanded }),
+    shallow
+  )
+  const { isSellMode, resetSellAssets, setIsSellMode } = useSellAsset(
```
Comment (vm) [inline src/components/NavBar/ShoppingBag.tsx:17]: same here, what is this shallow for?

Comment (JFrankfurt) [reply src/components/NavBar/ShoppingBag.tsx:17]: image
https://github.com/pmndrs/zustand/blob/d27ea948843b2c955ce2be530e5e963dff8958de/readme.md

shallow primitive value equality--won't work with deeply nested state, iirc

File: src/components/NavBar/ShoppingBag.tsx (hunk 1 above)
Comment (JFrankfurt) [reply src/components/NavBar/ShoppingBag.tsx:13]: zustand state read/write util `useBag` updated to provide this component with both current state and a write method (so can retain toggling behavior, and add conditional behavior based on current state.

**Review by vm — APPROVED** [review]

**Pull request merged**